src/providers/completer/citation.ts

```diff
@@ -184,12 +212,12 @@ export class Citation implements IProvider {
         const configuration = vscode.workspace.getConfiguration('latex-workshop')
         if (fs.statSync(file).size >= (configuration.get('intellisense.citation.maxfilesizeMB') as number) * 1024 * 1024) {
             this.extension.logger.addLogMessage(`Bib file is too large, ignoring it: ${file}`)
-            if (file in this.bibEntries) {
-                delete this.bibEntries[file]
+            if (this.bibEntries.has(file)) {
+                this.bibEntries.delete(file)
```

Comment:
The `.delete` method can be safely called without checking if the key is in the `Map`.